ports/rp2/machine_timer.c

```diff
@@ -57,7 +57,8 @@ STATIC int64_t alarm_callback(alarm_id_t id, void *user_data) {
 STATIC void machine_timer_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     machine_timer_obj_t *self = MP_OBJ_TO_PTR(self_in);
     qstr mode = self->mode == TIMER_MODE_ONE_SHOT ? MP_QSTR_ONE_SHOT : MP_QSTR_PERIODIC;
-    mp_printf(print, "Timer(mode=%q, period=%u, tick_hz=1000000)", mode, self->delta_us);
+    mp_printf(print, "Timer(mode=%q, period=%lu, tick_hz=1000000)",
+        mode, (uint32_t)(self->delta_us / 1000));
```

Comment:
I don't see how `%lu` is different to `%u`... on a 32-bit system it makes no difference (at least not in uPy's implementation).  Probably it needs `%llu` for a 64-bit number.

Also, because `tick_hz` is specified as 1_000_000, that means the period is in microseconds.  So it's actually correct as it stands.

Maybe there needs to be two variations of the print: one when the period is relatively small, and so given in us, and then one when it's relatively large and so given in ms.

Reply:
The major change was converting the delta_us to ms and casting it to uint32_t.

Reply:
The PR is updated to show the time as either µs or ms, depending on whether the period is longer or smaller than 1 Minute. The displayed ticks_hz quantity is changed accordingly.

Reply:
Thanks, looks good now.